Controller/ChangePasswordController.php

```diff
@@ -31,8 +31,19 @@
  * @author Thibault Duplessis <thibault.duplessis@gmail.com>
  * @author Christophe Coevoet <stof@notk.org>
  */
-class ChangePasswordController extends Controller
+class ChangePasswordController extends AbstractController
```

Comment:
why extend the `AbstractController` as you are injecting the services ?

Reply:
We're still using some of the shortcut methods - e.g. `generateUrl()`.  And those are more than just shortcuts - using the shortcut methods gives you lazy services - e.g. no need to inject a `Router` service if only 1 of 4 method needs it. By using `generateUrl()` it's only loaded when needed. Ok, router is a bad example - but that's the idea :)

Reply:
How about injecting `lazy` Router service then ?

Instead of making this controller as hybrid, isn't it better to make all dependencies explicit ? 
just curious.